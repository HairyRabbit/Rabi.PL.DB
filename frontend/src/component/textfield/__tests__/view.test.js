// -*- mode: js-jsx -*-
// -*- coding: utf-8 -*-

import React from 'react'
import { createStore } from 'redux'
import { Provider, connect } from 'react-redux'
import renderer from 'react-test-renderer'
import { shallow, mount, render } from 'enzyme'
import TextField from '../'
import {
  initModel,
  initAutoComplete,
  update,
  change,
  turn,
  toggle
} from '../state'

describe('test TextField', () => {
  const store = createStore(update, {
    value: '',
    autocomplete: {
      list: ['foo', 'bar', 'baz'],
      display: []
    }
  })
  const mapStateToProps = state => ({
    value: state.value,
    autocomplete: state.autocomplete
  })
  const mapDispatchToProps = dispatch => ({
    boundChange: type => value => dispatch(change(value, type))
  })
  const ConnectTextField = connect(mapStateToProps, mapDispatchToProps)(
    TextField
  )
  let component

  it('should init component.', () => {
    component = renderer.create(
      <Provider store={store}>
        <ConnectTextField />
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should change value on dispatched.', () => {
    store.dispatch(change('foo', 'text'))
    component = renderer.create(
      <Provider store={store}>
        <ConnectTextField />
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should clean value.', () => {
    component = renderer.create(
      <Provider store={store}>
        <ConnectTextField />
      </Provider>
    )
    component.toJSON().children[1].props.onKeyDown({
      preventDefault: jest.fn(),
      ctrlKey: true,
      which: 76
    })
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should tap tab key autocomplate matched value.', () => {
    store.dispatch(change('b'))
    component = renderer.create(
      <Provider store={store}>
        <ConnectTextField autocompleteValueDecode={x => x.value} />
      </Provider>
    )
    component.toJSON().children[1].props.onKeyDown({
      preventDefault: jest.fn(),
      which: 9
    })
    expect(component.toJSON()).toMatchSnapshot()
  })
})
