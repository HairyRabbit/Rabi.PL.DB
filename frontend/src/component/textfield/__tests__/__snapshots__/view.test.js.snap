// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test TextField should change value on dispatched. 1`] = `
<div
  className="container"
>
  <div
    className="left"
  >
    <div
      className="circle"
    />
  </div>
  <input
    className="main"
    id={undefined}
    name={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    type="text"
    value="foo"
  />
</div>
`;

exports[`test TextField should clean value. 1`] = `
<div
  className="container"
>
  <div
    className="left"
  >
    <div
      className="circle"
    />
  </div>
  <input
    className="main"
    id={undefined}
    name={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`test TextField should init component. 1`] = `
<div
  className="container"
>
  <div
    className="left"
  >
    <div
      className="circle"
    />
  </div>
  <input
    className="main"
    id={undefined}
    name={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`test TextField should tap tab key autocomplate matched value. 1`] = `
<div
  className="container"
>
  <div
    className="left"
  >
    <div
      className="circle"
    />
  </div>
  <input
    className="main"
    id={undefined}
    name={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    type="text"
    value="bar"
  />
</div>
`;
